compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1063,14 +1092,21 @@ impl<'a> Context<'a> {
         dfg: &DataFlowGraph,
         index: ValueId,
         store_value: Option<ValueId>,
+        offset: usize,
     ) -> Result<(AcirVar, Option<AcirValue>), RuntimeError> {
         let (array_id, array_typ, block_id) = self.check_array_is_initialized(array, dfg)?;
 
         let index_var = self.convert_numeric_value(index, dfg)?;
         let index_var = self.get_flattened_index(&array_typ, array_id, index_var, dfg)?;
 
-        let predicate_index =
-            self.acir_context.mul_var(index_var, self.current_side_effects_enabled_var)?;
+        let predicate_index = if offset != 0 {
+            let offset = self.acir_context.add_constant(offset);
+            let sub = self.acir_context.sub_var(index_var, offset)?;
```

Comment:
Could we add some comments as to why we are subtracting the offset to multiply by the predicate and then adding the offset back?

Reply:
there is already a comment:
`if an offset is specified, predicate_index is index*predicate + (1-predicate)*offset`
do you want more explanation?

Reply:
Ah I see now that is fine then